src/lib/y2users/linux/writer.rb

```diff
@@ -124,15 +130,38 @@ def write
       CHPASSWD = "/usr/sbin/chpasswd".freeze
       private_constant :CHPASSWD
 
+      # Whether given user does not exist yet
+      #
+      # FIXME: choose a better criteria to identify a new user. An internal Y2User::User id?
+      #
+      # @param user [Y2User::User]
+      # @return [Boolean] true if there is a user with same name in @initial_config; false otherwise
+      def new_user?(user)
+        @initial_users_names ||= initial_config.users.map(&:name)
+
+        !@initial_users_names.include?(user.name)
+      end
+
+      # Executes the command for creating the user
+      #
+      # @param user [Y2User::User]
       def add_user(user)
+        return unless new_user?(user)
+
         Yast::Execute.on_target!(USERADD, *useradd_options(user))
+      rescue Cheetah::ExecutionFailed => e
+        log.error("Error creating user '#{user.name}' - #{e.message}")
+      end
+
+      # Executes the command for setting the password of given user
+      #
+      # @param user [Y2User::User]
+      def change_password(user)
+        return unless user.password&.value
+
```

Comment:
I see here big three TODOs that I would like to have mentioned in code ( I know none of it is needed for installation ):

1. user modification
2. other password configuration
3. user remove

Reply:
They are somehow stated few lines above, on the user class documentation. Anyway, I'll recheck them tomorrow. Thanks!